python/tank/platform/software_launcher.py

```diff
@@ -234,6 +234,21 @@ def shotgun(self):
         """
         return self.sgtk.shotgun
 
+    def _get_minimum_supported_version(self):
+        """
+        The minimum software version that is supported by the launcher.
+        Returned as a string, for example "2015" or "2015.3.sp3".
+        Returns ``None`` if no constraint has been set.
+        Also see related method :meth:`~is_version_supported`.
+        """
+        return self.__minimum_supported_version
+
+    def _set_minimum_supported_version(self, version):
```

Comment:
Maybe I'm taking a bazooka here to fix this issue, but maybe the `create_engine_launcher` could take `args` and  `kwargs` parameters that are passed down to the `__init__` of the software launcher, which would allow you to pass a version through when instantiating your software launcher while unit testing.

Or if you won't want to, maybe you can mock `_get_minimum_supported_version` so the property taps into that instead?